src/main/java/com/celements/mandatory/AddDateIDXAsIndexToRecycleBin.java

```diff
@@ -0,0 +1,45 @@
+package com.celements.mandatory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+
+import com.celements.query.IQueryExecutionServiceRole;
+import com.xpn.xwiki.XWikiException;
+
+@Component("celements.mandatory.addIndexToRecycleBin")
+public class AddDateIDXAsIndexToRecycleBin implements IMandatoryDocumentRole {
```

Comment:
Confusing naming, contains too much detail. We recently simply named these kind of mandatory documents `...Indexes`, e.g. `CelementsIndexes`. Perhaps `XWikiRecycleBinIndexes` would suffice here?